src/Workleap.Extensions.Mongo/Indexing/IndexDeleter.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Workleap.Extensions.Mongo.Indexing;
+
+/// <summary>
+/// Delete indexes for a particular collection by comparing existing indexes in the database
+/// with the expected indexes declared in the code.
+/// </summary>
+internal sealed class IndexDeleter
+{
+    private readonly IMongoDatabase _database;
+    private readonly ILogger _logger;
+    
+    public IndexDeleter(IMongoDatabase database, ILogger<IndexDeleter> logger)
+    {
+        this._database = database;
+        this._logger = logger;
+    }
+    
+    public async Task ProcessAsync(Dictionary<string, IList<UniqueIndexName>> expectedIndexes, CancellationToken cancellationToken)
+    {
+        foreach (var expectedIndexForCollection in expectedIndexes)
+        {
+            var existingIndexesForCollection = await this.GetExistingIndexes(expectedIndexForCollection.Key).ConfigureAwait(false);
```

Comment:
GetExistingIndexes seems to get indexes for a collection, shouldn't it only be done once and not for every expected index? We could keep an collection internally and update it aswell with what we decide to remove instead of reaching out to the DB every time no?